### PR TITLE
docs: Move build system examples to new file

### PR DESCRIPTION
**Pull request by prince-chrismc**
This is a pretty straight forward move, I am trying to remove some of the clutter from the README for the next installment of 
 #13673 

As this is a simple move I am not included to change the wording too much, this is weaker section and most of the examples are out of date. Future work on this one.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
Putting the folders back in https://github.com/conan-io/conan-center-index/pull/14858

File: docs/error_knowledge_base.md [hunk 1]
```diff
@@ -445,7 +445,7 @@ class SomeRecipe(ConanFile):
 
 There is the case when the package is header-only, but the options affects the generated artifact, (e.g. kanguru, pagmo2 ...), so you need to use `self.info.settings.clear()` instead.
 
-- For "tool" recipes ([example](https://github.com/conan-io/conan-center-index/blob/e604534bbe0ef56bdb1f8513b83404eff02aebc8/recipes/cmake/3.x.x/conanfile.py#L104-L105)) which only provide binaries, see [our packing policy](adding_packages/conanfile_attributes.md#settings) for more, should do as follows:
+- @prince-chrismc This needs to be better --For "tool" recipes ([example](https://github.com/conan-io/conan-center-index/blob/e604534bbe0ef56bdb1f8513b83404eff02aebc8/recipes/cmake/3.x.x/conanfile.py#L104-L105)) which only provide binaries, see [our packaging policy](adding_packages/build_and_package.md) for more, should do as follows:
```
Comment (SSE4) [inline docs/error_knowledge_base.md:448]: leftover?

File: docs/adding_packages/build_and_package.md [hunk 2]
```diff
@@ -34,3 +40,41 @@ Both methods often use build helpers to build binaries and install them into the
...
+## Build System Examples
+
+The [Conan's documentation](https://docs.conan.io) is always a good place for technical details.
+General patterns about how they can be used for OSS in ConanCenterIndex can be found in the
+[package templates](../package_templates/README.md) sections. These are excellent to copy and start from.
+
+### Header Only
+
+If you are looking for header-only projects, you can take a look on [header-only template](../package_templates/header_only).
+Also, Conan Docs has a section about [how to package header-only libraries](https://docs.conan.io/en/latest/howtos/header_only.html).
```
Comment (SSE4) [inline docs/adding_packages/build_and_package.md:53]:
```suggestion
Also, Conan Docs have a section about [how to package header-only libraries](https://docs.conan.io/en/latest/howtos/header_only.html).
```

File: docs/adding_packages/build_and_package.md [hunk 3]
```diff
@@ -34,3 +40,41 @@ Both methods often use build helpers to build binaries and install them into the
...
+
+### CMake
+
+For C/C++ projects which use CMake for building, you can take a look on [cmake package template](../package_templates/cmake_package).
+
+Another common use case for CMake based projects, both header only and compiled, is _modeling components_ to match the `find_package` and export the correct targets from Conan's generators. A basic examples of this is [cpu_features](https://github.com/conan-io/conan-center-index/blob/master/recipes/cpu_features/all/conanfile.py), a moderate/intermediate example is [cpprestsdk](https://github.com/conan-io/conan-center-index/blob/master/recipes/cpprestsdk/all/conanfile.py), and a very complex example is [OpenCV](https://github.com/conan-io/conan-center-index/blob/master/recipes/opencv/4.x/conanfile.py).
```
Comment (SSE4) [inline docs/adding_packages/build_and_package.md:59]:
```suggestion
Another common use case for CMake-based projects, both header only and compiled, is _modeling components_ to match the `find_package` and export the correct targets from Conan's generators. A basic examples of this is [cpu_features](https://github.com/conan-io/conan-center-index/blob/master/recipes/cpu_features/all/conanfile.py), a moderate/intermediate example is [cpprestsdk](https://github.com/conan-io/conan-center-index/blob/master/recipes/cpprestsdk/all/conanfile.py), and a very complex example is [OpenCV](https://github.com/conan-io/conan-center-index/blob/master/recipes/opencv/4.x/conanfile.py).
```

File: docs/adding_packages/build_and_package.md [hunk 4]
```diff
@@ -34,3 +40,41 @@ Both methods often use build helpers to build binaries and install them into the
...
+
+### Autotools
+
+There is an [autotools package template](../package_templates/autotools_package/) amiable to start from.
+
+However, if you need to use autotools for building, you can take a look on [libalsa](https://github.com/conan-io/conan-center-index/blob/master/recipes/libalsa/all/conanfile.py), [kmod](https://github.com/conan-io/conan-center-index/blob/master/recipes/kmod/all/conanfile.py), [libcap](https://github.com/conan-io/conan-center-index/blob/master/recipes/libcap/all/conanfile.py).
+
+Many projects offer **pkg-config**'s `*.pc` files which need to be modeled using components. A prime example of this is [Wayland](https://github.com/conan-io/conan-center-index/blob/master/recipes/wayland/all/conanfile.py).
```
Comment (SSE4) [inline docs/adding_packages/build_and_package.md:67]:
```suggestion
Many projects offer [**pkg-config**'s](https://www.freedesktop.org/wiki/Software/pkg-config/) `*.pc` files which need to be modeled using components. A prime example of this is [Wayland](https://github.com/conan-io/conan-center-index/blob/master/recipes/wayland/all/conanfile.py).
```

**Review by RubenRBS — DISMISSED** [review]
👍 other than the typos @SSE4 found

File: docs/error_knowledge_base.md (hunk 1 above)
Comment (prince-chrismc) [reply docs/error_knowledge_base.md:448]: Making it obvious these are out-of-date for the reader :)

Comment (prince-chrismc) [reply docs/error_knowledge_base.md:448]:
```suggestion
- @prince-chrismc This needs to a better example; For "tool" recipes ([example](https://github.com/conan-io/conan-center-index/blob/e604534bbe0ef56bdb1f8513b83404eff02aebc8/recipes/cmake/3.x.x/conanfile.py#L104-L105)) which only provide binaries, see [our packaging policy](adding_packages/build_and_package.md) for more, should do as follows:
```

**Review by SSE4 — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**